internal/command/deploy/machines.go

```diff
@@ -383,12 +386,31 @@ func (md *machineDeployment) latestImage(ctx context.Context) (string, error) {
 	return resp.App.CurrentReleaseUnprocessed.ImageRef, nil
 }
 
-func (md *machineDeployment) setMachineGuest(vmSize string) error {
-	if vmSize == "" {
-		return nil
-	}
+func (md *machineDeployment) setMachineGuest(vmSize string, vmCPUKind string, vmCPUs int, vmMem int) error {
 	md.machineGuest = &api.MachineGuest{}
-	return md.machineGuest.SetSize(vmSize)
+	if vmSize != "" {
+		if err := md.machineGuest.SetSize(vmSize); err != nil {
+			return err
+		}
+	}
+
+	if vmCPUKind == "" {
+		md.machineGuest.CPUKind = "shared"
+	} else {
+		if md.machineGuest.CPUKind == "shared" || md.machineGuest.CPUKind == "performance" {
+			md.machineGuest.CPUKind = vmCPUKind
+		} else {
+			return fmt.Errorf("invalid machine CPU kind requested, '%s', expected to start with 'shared' or 'performance'", vmCPUKind)
+		}
+	}
```

Comment:
this whole block is awkward 🤔 

I was expecting it to:
1. Override `md.machineGuest.CPUKind` only if vmCPUKind is not an empty string
2. Default to `shared` only `md.machineGuest.CPUKind` is not set
3. (Optionally) fail if vmCPUKind is not one of `[]string{"shared", "performance"}` 

as it currently reads, it does a few unexpected things:
1. Sets `md.machineGuest.CPUKind` to "shared" if --vm-cpu-kind is not passed but overrides what `--vm-size` could have set
2. Checks `md.machineGuest.CPUKind` is one of "shared" or "performance", but doesn't check the value of `--vm-cpu-kind` is valid.

Reply:
Another good catch! This should've been caught by my tests, but those also had a bug :/.